enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,504 @@
+
+**orchestration tool** is intended to be a software component provided
+by the end user, but the steps may also be performed by **cluster
+creator** if no software component is available.
+
+**installer** is the OpenShift installer binary, `openshift-install`.
+
+**node0** is the first host to be added to a cluster.
+
+### Steps for Deploying a Disconnected Cluster
+
+1. The infrastructure admin launches an image registry visible to the
+   hosts that will form the cluster and to a host that has internet
+   access so the release image can be downloaded.
+2. The cluster creator obtains the `oc` command line tool and uses `oc`
+   to copy the OpenShift release image into the local image registry
+   in one of the usual ways with [`oc adm
+   mirror`](https://docs.openshift.com/container-platform/4.10/installing/disconnected_install/installing-mirroring-installation-images.html)
+   or [the `oc mirror`
+   plugin](https://docs.openshift.com/container-platform/4.10/installing/disconnected_install/installing-mirroring-disconnected.html).
+3. The cluster creator obtains the OpenShift installer in one of [the
+   usual
+   ways](https://docs.openshift.com/container-platform/4.9/installing/installing_bare_metal/installing-bare-metal.html#installation-obtaining-installer_installing-bare-metal).
+4. The cluster creator triggers their orchestration tool to run the
+   installer.
+5. The orchestration tool prepares an `install-config.yaml`, including
+   `NMState` content to configure the static network settings for the
+   hosts, using input data and templates.
+6. The orchestration tool runs `openshift-install create image`.
+7. The installer extracts a copy of the RHCOS ISO from the OpenShift
+   release payload.
+8. The installer combines the inputs provided by the user and produces
+   an Ignition config.
+9. The installer combines the Ignition config and the RHCOS ISO to
+   create the bootable image configured to run the assisted service or
+   agent based on the host that boots the image and writes the new
+   image and cluster credentials to disk.
+10. The orchestration tool copies the ISO to a location that the hosts
+    or VMs can boot from it. For bare metal, that will generally be a
+    web server visible to the management controllers in the bare metal
+    hosts (because the management controllers are unlikely to be on
+    the same network as the external interfaces for the cluster
+    hosts). For vSphere or other hypervisors, that may be a filesystem
+    on the hypervisor host.
+11. The orchestration tool configures the host to boot the ISO. For
+    bare metal, that means configuring the baseboard management
+    controller (BMC) in each host to expose the ISO through the
+    virtual media interface. For vSphere or other hypervisors, similar
+    settings of the VM need to be adjusted.
+12. The orchestration tool boots the hosts.
+13. The orchestration tool uses `openshift-install wait-for
```

Comment:
will it be modified to work with assisted reset api on the first stage and then move to kube?

Reply:
Yes, see the discussion around line 203-217 in the latest draft.